### PR TITLE
Added --refcase argument to summary_resample

### PR DESCRIPTION
**Pull request by joakim-hove**
**Task**
Added `--refcase=xxx` argument to the summary_resample script.

[x]: Statoil tests pass locally

### CODE REVIEW

File: lib/vector_template.h.in [hunk 1]
```diff
@@ -117,6 +117,8 @@ typedef @TYPE@ (@TYPE@_ftype) (@TYPE@);
   void                 @TYPE@_vector_fwrite_data( const @TYPE@_vector_type * vector , FILE * stream );
   void                 @TYPE@_vector_fread_data( @TYPE@_vector_type * vector , int size, FILE * stream);
   bool                 @TYPE@_vector_equal(const @TYPE@_vector_type * vector1 , const @TYPE@_vector_type * vector2);
+  int                  @TYPE@_vector_first_equal(const @TYPE@_vector_type * vector1, const @TYPE@_vector_type * vector2, int offset);
+  int                  @TYPE@_vector_first_not_equal(const @TYPE@_vector_type * vector1, const @TYPE@_vector_type * vector2, int offset);
```
Comment (pgdr) [inline lib/vector_template.h.in]: This went a bit under the radar ... why are these two functions added?

Comment (joakim-hove) [reply lib/vector_template.h.in]: They were utility functions to be able to write code of the type:
```python
if not v1 == v2:
     print "Vectors are not equal - first difference in index: %d" % v1.first_different( v2 )
```

Comment (pgdr) [reply lib/vector_template.h.in]: I would say, but you probably already know that, it was a bad idea, and leads to just more exposed symbols that we now need to maintain in the future...  Does this belong in `ecl`?  No.

If we absolutely need to implement and maintain our own "standard vector" class, we should look to `std::vector`.  That is the absolute maximum API we should expose, everything else should be caller's responsibility.

Implementing "first equal" would be trivial to do in the calling scope in Python.  If that somehow turned out to be too slow, _then_ we could consider implementing a helper function that could do it.

Now it's just yet another symbol we expose and lose control over.

Comment (joakim-hove) [reply lib/vector_template.h.in]: Well - the PR was open for 11 days; I specifically asked for review at one stage, with zero response - I do find it slightly frustrating that you see it as a valuable contribution to come with this type of comment after the PR has been merged.

Mainly due to one of your comments from October and these two issues: https://github.com/Statoil/libecl/issues/290 and https://github.com/Statoil/libecl/issues/27 libecl is seeing both a steady stream of feature move/remove PR's and also a large reorganization; although this PR contained a change in what you perciieve as the wrong direction. I hope you can acknowledge that your input and opinions already influence the direction of libecl quite significantly.

**Pull request merged**